src/components/AddTask/AddTask.tsx

```diff
@@ -0,0 +1,82 @@
+import "./AddTask.css";
+import addButton from "../../assets/add-button.svg";
+import TaskListContainer from "../Task/TaskList/TaskListContainer";
+import { useForm, SubmitHandler } from "react-hook-form";
+import { yupResolver } from "@hookform/resolvers/yup";
+import * as yup from "yup";
+
+interface Task {
+  id: number;
+  name: string;
+  completed: boolean;
+}
+
+interface AddTaskProps {
+  tasks: Task[];
+  addTask: (task: Task) => void;
+}
+
+const schema = yup
+  .object({
+    taskName: yup
+      .string()
+      .required("Task name is required")
+      .min(3, "Task name must be at least 3 characters long"),
+  })
+  .required();
+
+const AddTask = ({ tasks, addTask }: AddTaskProps): JSX.Element => {
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
```

Comment:
unable to find where are we using hook-form errors 
can we use this?

Reply:
yes we can. I did not think it was necessary to display errors however I did need the reset . The problem was when I removed {errors} it was giving error on the reset argument in formState. That is why I kept errors and did not remove it. However Errors can be used to display errors in the task names etc